basic_nodes/src/main/java/club/mondaylunch/gatos/basicnodes/ListSetOperationNodeType.java

```diff
@@ -0,0 +1,97 @@
+package club.mondaylunch.gatos.basicnodes;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
+import club.mondaylunch.gatos.core.data.DataBox;
+import club.mondaylunch.gatos.core.data.DataType;
+import club.mondaylunch.gatos.core.data.ListDataType;
+import club.mondaylunch.gatos.core.graph.connector.NodeConnector;
+import club.mondaylunch.gatos.core.graph.type.NodeType;
+
+public class ListSetOperationNodeType extends NodeType.Process {
+    private static final DataType<SetOperation> SET_OPERATION = DataType.register("set_operation", SetOperation.class);
+
+    @Override
+    public Map<String, DataBox<?>> settings() {
+        return Map.of(
+            "set_operation", SET_OPERATION.create(SetOperation.UNION)
+        );
+    }
+
+    @Override
+    public Set<NodeConnector.Input<?>> inputs(UUID nodeId, Map<String, DataBox<?>> settings, Map<String, DataType<?>> inputTypes) {
+        return Set.of(
+            new NodeConnector.Input<>(nodeId, "list_first", ListDataType.GENERIC_LIST),
+            new NodeConnector.Input<>(nodeId, "list_second", ListDataType.GENERIC_LIST)
+        );
+    }
+
+    @Override
+    public Set<NodeConnector.Output<?>> outputs(UUID nodeId, Map<String, DataBox<?>> settings, Map<String, DataType<?>> inputTypes) {
+        var outputType = this.getOutputListTypeOrThrow(inputTypes, "list_first", "list_second");
+        return Set.of(
+            new NodeConnector.Output<>(nodeId, "output", outputType));
+    }
+
+    @Override
+    public Map<String, CompletableFuture<DataBox<?>>> compute(Map<String, DataBox<?>> inputs, Map<String, DataBox<?>> settings, Map<String, DataType<?>> inputTypes) {
+        var outputType = this.getOutputListTypeOrThrow(inputTypes, "list_first", "list_second");
+        var currentOp = DataBox.get(settings, "set_operation", SET_OPERATION).orElseThrow();
+        List<Object> first = new ArrayList<>(DataBox.get(inputs, "list_first", ListDataType.GENERIC_LIST).orElseThrow().stream().distinct().toList());
```

Comment:
It's probably better if you put the contents of the `list_first` and `list_second` databoxes into hash sets, operate on those using `Set#addAll` for the union, `Set#retainAll` for the intersection and `Set#removeAll` for the difference, and then put the set contents back into a list.